### PR TITLE
improve the necessary input arguments checking for python

### PR DESCRIPTION
**Pull request by dreamryx**
**What this PR does / why we need it**:
Add some code for checking the necessary input arguments in python sdk.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/metadata/122)

### CODE REVIEW

**Comment by dreamryx** [comment]
@zhenghuiwang Please help review it. Thanks. :) 
By the way, do you think there is a need to write some wrapper api in metadata.py to help user create artifactType and so on easily. Please refer to #120.

File: sdk/python/kfmd/metadata.py [hunk 1]
```diff
@@ -46,6 +46,10 @@ def __init__(self,
       labels {object} Optional key/value string pairs to label the workspace.
     """
     # TODO(zhenghuiwang): check each field's type and whether set.
```
Comment (zhenghuiwang) [inline sdk/python/kfmd/metadata.py]: Can you remove the this TODO?

Same below.

File: sdk/python/kfmd/metadata.py [hunk 2]
```diff
@@ -46,6 +46,10 @@ def __init__(self,
...
+    if backend_url_prefix is None and type(backend_url_prefix) != str:
+          raise ValueError("'backend_url_prefix' must be set and in string type.")
+    if name is None and type(name) != str:
```
Comment (zhenghuiwang) [inline sdk/python/kfmd/metadata.py]: it should be `if name is None or type(name) != str`, not `and`.

Same below.

File: sdk/python/kfmd/metadata.py [hunk 3]
```diff
@@ -46,6 +46,10 @@ def __init__(self,
       labels {object} Optional key/value string pairs to label the workspace.
     """
     # TODO(zhenghuiwang): check each field's type and whether set.
+    if backend_url_prefix is None and type(backend_url_prefix) != str:
```
Comment (zhenghuiwang) [inline sdk/python/kfmd/metadata.py]: Could you add some unit tests for this?

File: sdk/python/kfmd/metadata.py (hunk 2 above)
Comment (dreamryx) [reply sdk/python/kfmd/metadata.py]: Yes. I will change it now.

File: sdk/python/kfmd/metadata.py (hunk 3 above)
Comment (dreamryx) [reply sdk/python/kfmd/metadata.py]: Yes. Of course.

File: sdk/python/kfmd/metadata.py (hunk 2 above)
Comment (dreamryx) [reply sdk/python/kfmd/metadata.py]: Thanks for your review!

File: sdk/python/kfmd/metadata.py (hunk 1 above)
Comment (dreamryx) [reply sdk/python/kfmd/metadata.py]: OK. Got it.

**Comment by zhenghuiwang** [comment]
Thanks!

/lgtm
/approve

**Comment by dreamryx** [comment]
@zhenghuiwang   Thanks for your review!

**Pull request merged**